Дипломный проект. Этап 2. Основной сервис.

ViktorSalk opened the pull request:
1) Создал ветку main_svc.
2) Обновил postman.

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.
3) Доработал модуль ewm-main-service.

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.
3) Доработал модуль ewm-main-service.
4) Доработал модуль ewm-stat-service.

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.
3) Доработал модуль ewm-main-service.
4) Доработал модуль ewm-stat-service.
5) Обновил в модуле ewm-stat-service, stat_server.
6) Обновил основной докер и пом.

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.
3) Доработал модуль ewm-main-service.
4) Доработал модуль ewm-stat-service.
5) Обновил в модуле ewm-stat-service, stat_server.
6) Обновил основной докер и пом.
7) Обновил тесты.

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.
3) Доработал модуль ewm-main-service.
4) Доработал модуль ewm-stat-service.
5) Обновил в модуле ewm-stat-service, stat_server.
6) Обновил основной докер и пом.
7) Обновил тесты.

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.
3) Доработал модуль ewm-main-service.
4) Доработал модуль ewm-stat-service.
5) Обновил в модуле ewm-stat-service, stat_server.
6) Обновил основной докер и пом.
7) Обновил тесты.
8) Добавил новые тесты.

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.
3) Доработал модуль ewm-main-service.
4) Доработал модуль ewm-stat-service.
5) Обновил в модуле ewm-stat-service, stat_server.
6) Обновил основной докер и пом.
7) Обновил тесты.
8) Добавил новые тесты.

VadimZharkov commented on ewm-main-service/src/main/java/ru/practicum/ewm/controller/priv/event/EventPrivateController.java:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    @GetMapping("/{eventId}/requests")
+    public List<ParticipationRequestDto> getAllRequestByEventFromOwner(@PathVariable(value = "userId") @Min(1) Long userId,
```
Если имя переменной пути совпадает с именем метода, то value =  излишне

ViktorSalk replied:
Убрал избыточный атрибут value.

VadimZharkov commented on ewm-main-service/src/main/java/ru/practicum/ewm/model/User.java:
```diff
@@ -0,0 +1,25 @@
+import lombok.NoArgsConstructor;
+
+@Entity(name = "users")
+@Data
```
`@Data` для сущностей не рекомендуется использовать в дипломной работе.
https://habr.com/ru/companies/haulmont/articles/564682/

ViktorSalk replied:
Заменил @Data на специальные аннотации: @Getter и @Setter для методов доступа, @toString.

VadimZharkov commented on ewm-main-service/src/main/java/ru/practicum/ewm/service/impl/EventServiceImpl.java:
```diff
@@ -0,0 +1,556 @@
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class EventServiceImpl implements EventService {
```
В классы-сервисы лучше добавить `@Transactional(readOnly = true)` и `@Transactional`.

ViktorSalk replied:
Спасибо), для надлежащей обработки трансакций добавил @Transactional(readOnly = true) и @Transactional.

VadimZharkov commented on ewm-main-service/src/main/java/ru/practicum/ewm/service/impl/EventServiceImpl.java:
```diff
@@ -0,0 +1,556 @@
+            hasChanges = true;
+        }
+        if (!hasChanges) {
+
```
лишняя пустая строка)

ViktorSalk replied:
Спасибо)

VadimZharkov commented on ewm-main-service/src/main/resources/schema.sql:
```diff
@@ -1,4 +1,76 @@
+    annotation         VARCHAR(2000)               NOT NULL,
+    category_id        BIGINT                      NOT NULL,
+    confirmed_Requests BIGINT,
+    create_date        TIMESTAMP WITHOUT TIME ZONE,
```
NOT NULL

ViktorSalk replied:
Добавил ограничение NOT NULL.

VadimZharkov commented on ewm-main-service/src/main/resources/schema.sql:
```diff
@@ -1,4 +1,76 @@
+    event_date         TIMESTAMP WITHOUT TIME ZONE NOT NULL,
+    initiator_id       BIGINT                      NOT NULL,
+    location_id        BIGINT,
+    paid               BOOLEAN,
```
NOT NULL

ViktorSalk replied:
Добавил ограничение NOT NULL.

VadimZharkov commented on ewm-main-service/src/main/resources/schema.sql:
```diff
@@ -1,4 +1,76 @@
+    id           BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    event_id     BIGINT NOT NULL,
+    requester_id BIGINT NOT NULL,
+    create_date  TIMESTAMP WITHOUT TIME ZONE,
```
NOT NULL

ViktorSalk replied:
Добавил ограничение NOT NULL.

VadimZharkov commented on ewm-main-service/src/main/resources/schema.sql:
```diff
@@ -1,4 +1,76 @@
+    event_id     BIGINT NOT NULL,
+    requester_id BIGINT NOT NULL,
+    create_date  TIMESTAMP WITHOUT TIME ZONE,
+    status       VARCHAR(20),
```
NOT NULL

ViktorSalk replied:
Добавил ограничение NOT NULL.

VadimZharkov commented on ewm-main-service/src/main/resources/schema.sql:
```diff
@@ -1,4 +1,76 @@
+    participant_limit  INTEGER DEFAULT 0,
+    published_date     TIMESTAMP WITHOUT TIME ZONE,
+    request_moderation BOOLEAN DEFAULT true,
+    status             VARCHAR(200),
```
NOT NULL

ViktorSalk replied:
Добавил ограничение NOT NULL.

VadimZharkov commented on ewm-main-service/src/main/resources/schema.sql:
```diff
@@ -1,4 +1,76 @@
+
+CREATE TABLE IF NOT EXISTS compilations_to_event
+(
+    id             BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
```
В таблицах-связках нет необходимости делать колонку с id.
Лучше сделать первичный ключ по двум колонкам
CONSTRAINT pk_compilation_to_events PRIMARY KEY (compilation_id, event_id)

ViktorSalk replied:
Убрал лишнее поле id из таблицы compilations_to_event и сделал составной первичный ключ по (event_id, compilation_id).

VadimZharkov commented on ewm-stat-service/stat-server/src/main/resources/schema.sql:
```diff
@@ -0,0 +1,10 @@
+    app     VARCHAR(255) NOT NULL,
+    uri     VARCHAR(255) NOT NULL,
+    ip      VARCHAR(255) NOT NULL,
+    created TIMESTAMP WITHOUT TIME ZONE
```
NOT NULL

ViktorSalk replied:
Добавил ограничение NOT NULL.

VadimZharkov reviewed (changes requested):
Хорошая работа, посмотри, что можно улучшить.

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.
3) Доработал модуль ewm-main-service.
4) Доработал модуль ewm-stat-service.
5) Обновил в модуле ewm-stat-service, stat_server.
6) Обновил основной докер и пом.
7) Обновил тесты.
8) Добавил новые тесты.
9) В ewm-stat-service и ewm-main-service в схемы добавил ограничения NOT NULL.
10) В ewm-main-service в схеме убрал лишнее поле id из таблицы compilations_to_event и сделал составной первичный ключ по (event_id, compilation_id).

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.
3) Доработал модуль ewm-main-service.
4) Доработал модуль ewm-stat-service.
5) Обновил в модуле ewm-stat-service, stat_server.
6) Обновил основной докер и пом.
7) Обновил тесты.
8) Добавил новые тесты.
9) В ewm-stat-service и ewm-main-service в схемы добавил ограничения NOT NULL.
10) В ewm-main-service в схеме убрал лишнее поле id из таблицы compilations_to_event и сделал составной первичный ключ по (event_id, compilation_id).
11) Для надлежащей обработки трансакций добавил @Transactional(readOnly = true) и @Transactional.

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.
3) Доработал модуль ewm-main-service.
4) Доработал модуль ewm-stat-service.
5) Обновил в модуле ewm-stat-service, stat_server.
6) Обновил основной докер и пом.
7) Обновил тесты.
8) Добавил новые тесты.
9) В ewm-stat-service и ewm-main-service в схемы добавил ограничения NOT NULL.
10) В ewm-main-service в схеме убрал лишнее поле id из таблицы compilations_to_event и сделал составной первичный ключ по (event_id, compilation_id).
11) Для надлежащей обработки трансакций добавил @Transactional(readOnly = true) и @Transactional.
12) Убрал избыточный атрибут value.

Viktor3642 committed:
Дипломный проект. Этап 2. Основной сервис.
1) Создал ветку main_svc.
2) Обновил postman.
3) Доработал модуль ewm-main-service.
4) Доработал модуль ewm-stat-service.
5) Обновил в модуле ewm-stat-service, stat_server.
6) Обновил основной докер и пом.
7) Обновил тесты.
8) Добавил новые тесты.
9) В ewm-stat-service и ewm-main-service в схемы добавил ограничения NOT NULL.
10) В ewm-main-service в схеме убрал лишнее поле id из таблицы compilations_to_event и сделал составной первичный ключ по (event_id, compilation_id).
11) Для надлежащей обработки трансакций добавил @Transactional(readOnly = true) и @Transactional.
12) Убрал избыточный атрибут value.
13) Заменил @Data на специальные аннотации: @Getter и @Setter для методов доступа, @toString.
